packages/framework-core/src/instrumentation/decorator/trace.ts

```diff
@@ -0,0 +1,58 @@
+import { TraceActionTypes, TraceParameters, TraceTypes, UUID } from '@boostercloud/framework-types'
+import { isTracerConfigured, notifyTrace } from '../trace-notifier'
+import { Booster } from '../../booster'
+
+export function Trace(actionType: TraceActionTypes = TraceActionTypes.CUSTOM, description?: string) {
+  return (target: unknown, member: string, descriptor: PropertyDescriptor): PropertyDescriptor => {
+    const originalMethod = descriptor.value
+    descriptor.value = async function (...args: Array<unknown>) {
+      const tracerConfigured = isTracerConfigured(actionType)
+      if (!tracerConfigured) {
+        return await originalMethod.apply(this, args)
+      }
+      const parameters = buildParameters(target, member, args, description, descriptor)
+      const startTime = new Date().getTime()
+      await notifyTrace(TraceTypes.START, actionType, parameters)
+      try {
+        return await originalMethod.apply(this, args)
+      } finally {
+        parameters.elapsedInvocationMillis = new Date().getTime() - startTime
+        await notifyTrace(TraceTypes.END, actionType, parameters)
+      }
+    }
+    return descriptor
+  }
+}
+
+function buildParameters(
+  target: unknown,
+  member: string,
+  args: unknown[],
+  description: string | undefined,
+  descriptor: PropertyDescriptor
+) {
+  const config = Booster.config
+  let internal = undefined
+  if (config && config.traceConfiguration.includeInternal) {
```

Comment:
Couldn't `traceConfiguration` be undefined?

Reply:
No, the `traceConfiguration` is set in the framework-types config class and it doesn't accepts undefined.